trl/ppo.py

```diff
@@ -260,6 +260,9 @@ def loss(self, old_logprobs, values, rewards, query, response, model_input):
 
         ratio = torch.exp(logprob - old_logprobs)
 
+        if len(ratio.shape) == 2:
```

Comment:
Why is that step needed now?

Reply:
using the latest version of `transformers`, it seems that `ratio` needs a thrid dimension otherwise the broadcasting op wont work when calling `*` in `pg_losses = -advantages * ratio` (for instance using gpt2,  `advantages.shape==(1, 9, 2)` and `ratio.shape == (1, 9)`). 
I got this issue with the latest versions of `transformers`, so I suspect that the way the outputs are dealt changed between the versions but didn't investigated further